Calculator.py

```diff
@@ -0,0 +1,134 @@
+import re
+import math
+from statistics import mean
+
+from pip._vendor.msgpack.fallback import xrange
+
+
+class Calculator:
+    def __init__(self, exercise):
+        self.__operations_regex_dict = {1: r'\(\d+[!%^+/*\-~$&@]\d+\)+',
+                                        2: r'\!',
+                                        3: r'[$&@]',
+                                        4: r'%',
+                                        5: r'\^',
+                                        6: r'[/*]',
+                                        7: r'[-+]'}
+        self.exercise = exercise.strip()

```

Comment:
consider a different name instead of exercise because as I read this I'm not sure what exercise means. Did you mean expression?